3p/ampcontext.js

```diff
@@ -29,6 +29,9 @@ export class AbstractAmpContext {
     dev().assert(!this.isAbstractImplementation_(),
         'Should not construct AbstractAmpContext instances directly');
 
+    /** @visibleForTesting */
+    this._ampInternalIsNewImpl = true;
```

Comment:
Since the test does the same check twice, once for the current implementation and another time for the replacement based only on an experiment flag, I'd like the test to ensure that it's actually testing for the new implementation.

Doing this made me find out that the experiment toggle in the test wasn't doing previously anything as it was set as `opt_transient`.

This can be cleaned up once the experiment flag is removed.